web/static/js/settings/rule-toggle.js

```
@@ -219,14 +219,41 @@
+    // Storage mirrors the checker's dynamic-default semantics (config replaces
+    // the whole RuleConfig; required_provider_ids is omitempty, so an omitted
+    // value persists as empty):
+    //   - all available checked  -> omit, storing empty so the rule keeps
+    //     requiring every available in-scope provider as they are added/removed.
+    //   - a strict subset checked -> store that subset, comma-separated.
+    //   - none checked            -> reject the save. An empty string means
+    //     "all", so silently storing it would invert the operator's intent;
+    //     require at least one provider (or disable the rule to require none).
+    var providerBoxes = form.querySelectorAll('input[type="checkbox"][name="required_provider_ids"]');
+    if (providerBoxes.length > 0) {
+      var checkedProviders = [];
```

Biome (2.5.3)

[error] This var should be declared at the root of the enclosing function.

(lint/correctness/noInnerDeclarations)